relation_calculator/relation_cal.c

```diff
@@ -0,0 +1,416 @@
+    lv_obj_set_size(g_rel_cal.ui.title, 760, 50);
+    lv_label_set_text(g_rel_cal.ui.title, "亲属关系计算器");
+    lv_obj_set_style_text_font(g_rel_cal.ui.title, &lv_font_chinese_siyuan_16, 0);
+    lv_obj_set_style_text_color(g_rel_cal.ui.title, lv_color_hex(0x2C3E50), 0);
+    lv_obj_set_style_text_align(g_rel_cal.ui.title, LV_TEXT_ALIGN_CENTER, 0);
+    lv_obj_set_style_pad_top(g_rel_cal.ui.title, 10, 0);
+    // 修改标题颜色为橙色
+    lv_obj_set_style_text_color(g_rel_cal.ui.title, lv_color_hex(0xFFA500), 0);
+
+    // 创建显示区域
+    g_rel_cal.ui.screen = lv_textarea_create(root);
+    lv_obj_set_size(g_rel_cal.ui.screen, LV_PCT(100), LV_PCT(40));
+    lv_obj_set_style_text_font(g_rel_cal.ui.screen, &lv_font_chinese_siyuan_16, 0);
+    lv_obj_set_style_bg_color(g_rel_cal.ui.screen, lv_color_hex(0x2C3E50), 0);
+    lv_obj_set_style_text_color(g_rel_cal.ui.screen, lv_color_white(), 0);
+    lv_obj_set_style_radius(g_rel_cal.ui.screen, 20, 0);
+    lv_obj_set_style_pad_all(g_rel_cal.ui.screen, 15, 0);
+    lv_textarea_set_cursor_click_pos(g_rel_cal.ui.screen, false);
+    lv_textarea_set_max_length(g_rel_cal.ui.screen, 128);
+    lv_textarea_set_align(g_rel_cal.ui.screen, LV_TEXT_ALIGN_LEFT);
+    lv_textarea_set_text(g_rel_cal.ui.screen, "");
+
+    // 创建按钮矩阵
+    g_rel_cal.ui.btnm = lv_btnmatrix_create(root);
+    lv_obj_set_style_border_width(g_rel_cal.ui.btnm, 0, 0);
+    lv_obj_set_size(g_rel_cal.ui.btnm, LV_PCT(100), LV_PCT(40));
+    lv_obj_set_style_text_font(g_rel_cal.ui.btnm, &lv_font_chinese_siyuan_16, LV_PART_MAIN);
+    lv_buttonmatrix_set_map(g_rel_cal.ui.btnm, btnm_map);
+    lv_obj_add_event_cb(g_rel_cal.ui.btnm, rel_cal_btnmatrix_event_cb, LV_EVENT_VALUE_CHANGED, root);
+    lv_obj_set_style_bg_color(g_rel_cal.ui.btnm, lv_color_hex(0xECF0F1), 0);
+    lv_obj_set_style_radius(g_rel_cal.ui.btnm, 10, 0);
+    lv_obj_set_style_pad_all(g_rel_cal.ui.btnm, 10, 0);
+    
+
+    // 创建提示信息
+    g_rel_cal.ui.note = lv_label_create(root);
+    lv_obj_set_size(g_rel_cal.ui.note, LV_PCT(100), LV_PCT(10));
+    lv_label_set_text(g_rel_cal.ui.note, "这是一个亲属关系计算器,这只是v1的简陋版。");
+    lv_obj_set_style_text_font(g_rel_cal.ui.note, &lv_font_chinese_siyuan_16, 0);
+    lv_obj_set_style_text_color(g_rel_cal.ui.note, lv_color_hex(0x7F8C8D), 0);
+    lv_obj_set_style_text_align(g_rel_cal.ui.note, LV_TEXT_ALIGN_CENTER, 0);
+}
+
+
+static void rel_cal_btnmatrix_event_cb(lv_event_t *e)
+{
+    lv_obj_t * obj = lv_event_get_target(e);
+    uint8_t id = lv_buttonmatrix_get_selected_button(obj);
+    const char * text = lv_buttonmatrix_get_button_text(obj, id);
+    rel_transform_handle(text);
+}
+
+static void rel_transform_handle(const char *text)
+{
+    static btnm_relation_t last_input = {"?", REL_NULL,GENDER_UNKNOWN};
+
+    if(g_rel_cal.status == STATUS_CALCULATED)
+    {
+        g_rel_cal.status = STATUS_INPUTTING;
+        lv_textarea_set_text(g_rel_cal.ui.screen,"");
+    }
+
+    if (0 == strcmp(text,"计算")) 
+    {
+        if(g_rel_cal.rel_count > 0)
```

Comment:
It is recommended to use enumeration or macros to handle button judgments. Using the same string in different places to determine whether it meets the logical requirements is neither elegant nor safe.